traced/tests/src/stopgap/guard_fail.rs

```diff
@@ -218,3 +220,45 @@ fn trace_looping_nested() {
     unsafe { ct.execute(&mut args) };
     assert_eq!(args.0, 100);
 }
+
+#[bench]
+fn bench_looping(b: &mut Bencher) {
+    struct InterpCtx {
+        compiled: bool,
+        count: u8
+    }
+
+    #[interp_step]
+    fn interp_step(io: &mut InterpCtx) -> bool {
+        if io.compiled {
+            if io.count < 255 {
+                io.count += 1;
+                false
+            } else {
+                true
+            }
+        } else {
+            if io.count < 255 {
+                io.count += 1;
+                false
+            } else {
+                true
+            }
+        }
+    }
+
+    let mut ctx = InterpCtx{compiled: false, count: 0};
+    #[cfg(tracermode = "hw")]
+    let th = start_tracing(TracingKind::HardwareTracing);
+    #[cfg(tracermode = "sw")]
+    let th = start_tracing(TracingKind::SoftwareTracing);
+    interp_step(&mut ctx);
+    let sir_trace = th.stop_tracing().unwrap();
+    let ct = compile_trace(sir_trace).unwrap();
+
+    b.iter(|| {
+        let mut args = InterpCtx{compiled: true, count: 0};
+        unsafe { ct.execute(&mut args) };
+        assert_eq!(args.count, 255);
+    });
```

Comment:
FWIW, this benchmark looks sensible to me.

Reply:
This does not benchmark the stopgap interpreter though. It's actually never run. This particular test `trace_looping_nested` just makes sure that "trace stitching" still works even if other functions return `true`.

The `recursion1/2` tests actually execute the stopgap interpreter, but it probably doesn't run long enough to bench it. This test is a better candidate for benchmarking the stopgap interpreter as it will fail a guard multiple times over: https://github.com/softdevteam/yk/blob/master/traced/ykrt/src/mt.rs#L1172.

Reply:
Aha, I see! OK, I think I can turn those into benchmarks.